Aptoide/aptoide-client-v8
# [feature/MOB-173] Theme settings

**What does this PR do?**

   This adds a theme option in Settings. I also did some minor refactors all around and bumped some dependencies.

**Note:** The strings are not yet correct, so do not accept.

**Database changed?**

  No

**How should this be manually tested?**

  Go to the settings and play around with the theme option and try every combination (system, light, dark) and see if it behaves as expected. Also check if it persists through killing and reopening the app.
  Note that it is expected that we cannot control the splashscreen color -- this is a known limitation and it is always dependent on the system default. It also happens with other apps with dark mode.

**What are the relevant tickets?**

  [MOB-173](https://aptoide.atlassian.net/browse/MOB-173)

**Code Review Checklist**
